### PR TITLE
Policy Error Fixes

### PR DESCRIPTION
**Pull request by wey-chiang**
### Background

@s0l0ist found 3 policies that were erroring out:

1. AWS.AccessKeys.AccountCreation
2. AWS.Config.GlobalResources
3. AWS.ELBv2.SSLPolicy

### Changes

- AWS.AccessKeys.AccountCreation
 - Wrapped the `datetime.strptime` in a try/except that attempts to use both the `PANTHER_TIME_FORMAT` and `AWS_TIME_FORMAT`
- AWS.Config.GlobalResources
 - Reworked the policy. Now checks if any of the recorders are configured for global resources on a per-account basis.
- AWS.ELBv2.SSLPolicy
 - Added conditionals for safe evaluation
 - Renamed iterator because of outer-scope shadow issue

### Testing

* `panther_analysis_tool test`
* Tested in live deployments with the resources causing issues created as unit tests

### CODE REVIEW

File: aws_config_policies/aws_config_global_resources.yml [hunk 1]
```diff
@@ -17,18 +17,56 @@ Runbook: >
...
+        "ResourceId": "123456789012::AWS.Config.Recorder.Meta",
+        "ResourceType": "AWS.Config.Recorder.Meta",
+        "Tags": null,
+        "Region": "global",
+        "Recorders": null,
+        "Name": "AWS.Config.Recorder.Meta",
+        "GlobalRecorderCount": 0
       }
+#  -
+#    Name: __REPLACE_ACCOUNT_ID_TO_SUPPORT_TESTING__ Global Resources Are Enabled
```
Comment (wey-chiang) [inline aws_config_policies/aws_config_global_resources.yml:35]: This name feels sketchy

File: aws_iam_policies/aws_access_keys_at_account_creation.py [hunk 2]
```diff
@@ -20,7 +19,10 @@ def policy(resource):
         return True
 
     create = resource["TimeCreated"]
-    key_rot_date = datetime.datetime.strptime(key_rot, AWS_TIME_FORMAT)
-    create_date = datetime.datetime.strptime(create, PANTHER_TIME_FORMAT)
+    key_rot_date = datetime.strptime(key_rot, AWS_TIME_FORMAT)
+    try:
+        create_date = datetime.strptime(create, PANTHER_TIME_FORMAT)
+    except ValueError:
+        create_date = datetime.strptime(create, AWS_TIME_FORMAT)
```
Comment (wey-chiang) [inline aws_iam_policies/aws_access_keys_at_account_creation.py:26]: Not entirely sure this is the best way to go about this...

File: aws_load_balancer_policies/aws_alb_ssl_policy.py [hunk 3]
```diff
@@ -10,9 +10,9 @@
 
 def policy(resource):
     # Ignore load balancers that aren't serving internet traffic
-    if resource["Scheme"] == "internal":
+    if resource.get("Scheme") == "internal":
         return True
 
-    return len(resource["Listeners"]) >= 1 and all(
-        (policy in TLS_1_2_POLICIES for policy in resource["SSLPolicies"].keys())
+    return len(resource.get("Listeners") if resource.get("Listeners") else []) >= 1 and all(
```
Comment (wey-chiang) [inline aws_load_balancer_policies/aws_alb_ssl_policy.py:16]: Might be a cleaner way to do this

File: aws_config_policies/aws_config_global_resources.yml [hunk 4]
```diff
@@ -17,18 +17,56 @@ Runbook: >
...
+#          "123456789012:us-west-2:AWS.Config.Recorder"
+#        ],
+#        "AccountId": "123456789012",
+#        "Name": "AWS.Config.Recorder.Meta",
+#        "GlobalRecorderCount": 1,
+#        "Tags": null,
+#        "Region": "global",
+#        "ResourceType": "AWS.Config.Recorder.Meta",
+#        "ResourceId": "123456789012::AWS.Config.Recorder.Meta"
+#      }
```
Comment (wey-chiang) [inline aws_config_policies/aws_config_global_resources.yml:72]: Really need unit test mocking for policies... 

Work is on the way though! The Unified Detection Pipeline should also go a long ways when it comes to discrepancies in things we can do in the rules engine but not in the policy engine

File: aws_config_policies/aws_config_global_resources.yml [hunk 5]
```diff
@@ -4,7 +4,7 @@ PolicyID: AWS.Config.GlobalResources
 DisplayName: AWS Config Global Resources
 Enabled: true
 ResourceTypes:
-  - AWS.Config.Recorder.Meta
+  - AWS.Config.Recorder
```
Comment (wey-chiang) [inline aws_config_policies/aws_config_global_resources.yml:7]:
```suggestion
  - AWS.Config.Recorder.Meta
```

File: aws_iam_policies/aws_access_keys_at_account_creation.py (hunk 2 above)
Comment (lindsey-w) [inline aws_iam_policies/aws_access_keys_at_account_creation.py:26]: could it also raise an exception in the except clause if it doesn't match AWS_TIME_FORMAT? I don't know if this is any better, but another option could be: 
```suggestion
        for format in [PANTHER_TIME_FORMAT, AWS_TIME_FORMAT]:
            try:
                create_date = datetime.strptime(create, format)
                break
            except ValueError:
                continue
```

🤷🏻‍♀️

File: aws_config_policies/aws_config_global_resources.yml (hunk 1 above)
Comment (lindsey-w) [reply aws_config_policies/aws_config_global_resources.yml:35]: Is this to alert someone that to uncomment they should replace the account id? seems reasonable to me

**Review by lindsey-w — APPROVED** [review]
👍🏻

File: aws_config_policies/aws_config_global_resources.yml (hunk 1 above)
Comment (wey-chiang) [reply aws_config_policies/aws_config_global_resources.yml:35]: yeah, exactly

File: aws_iam_policies/aws_access_keys_at_account_creation.py (hunk 2 above)
Comment (wey-chiang) [reply aws_iam_policies/aws_access_keys_at_account_creation.py:26]: I like this form better -- let's revisit it if timestamp issues become more prevalent.

**Pull request merged**